vitest.global-browser-setup.ts

```diff
@@ -0,0 +1,24 @@
+import { spawn } from 'node:child_process';
+
+export default async function setup() {
+  return new Promise((resolve, reject) => {
+    const cp = spawn('pnpm tsx packages/fuels/src/setup-launch-node-server.ts', {
+      detached: true,
+      shell: 'sh',
+    });
+
+    cp.stdout?.on('data', () => {
+      // return teardown function to be called when tests finish
+      // it will kill the server
+      resolve(() => {
+        // https://github.com/nodejs/node/issues/2098#issuecomment-169549789
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        process.kill(-cp.pid!);
+      });
+    });
+
+    cp.on('error', (err) => {
+      reject(err);
```

Comment:
We probably still want to kill the server on error, we could extract the `process.kill` to a function

```ts
  const killServer = () => {
      if (cp.pid) {
        process.kill(-cp.pid);
      }
    };
   ```
   
  and then call it on `error` and `exit` scenarios
  
  ```ts
  cp.stdout?.on('data', () => {
      // Return teardown function to be called when tests finish
      // It will kill the server
      resolve(killServer);
    });

    cp.on('error', (err) => {
      // Ensure server is killed if there's an error
      killServer();
      reject(err);
    });

    cp.on('exit', (code) => {
      if (code !== 0) {
        reject(new Error(`Server process exited with code ${code}`));
      }
    });
  ```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2378/commits/7a1ebbfe1a2944f5aebf3de9e75bd6003ba8b497